app/controllers/time_slot_controller.rb

```diff
@@ -17,6 +17,8 @@ def schedule
         return
       end
 
+      current_patient.appointments.where('start > ?', Time.zone.now).destroy_all
```

Comment:
```suggestion
      current_patient.appointments.destroy_all
```

Dessa forma não vai destruir só os agendamentos de um horário específico?

Acredito que a solução deveria ser excluir _todos_ os agendamentos daquele paciente e deixar apenas o novo. 

Ou a regra de negócio permite que um paciente tenha mais de um agendamento marcado?